### ISSUE TITLE
HDR dimming external display when turned on.

### ISSUE THREAD

**Issue by RoseAF1** [iina/iina#4979]
**System and IINA version:**

- macOS   Sonoma 14.5
- IINA  1.3.5 build 141

**Expected behavior:**
Play HDR video at expected brightness.

**Actual behavior:**
When HDR video is dragged to external HDR monitor, the brightness dims to an unusable level

**Steps to reproduce:**
While viewing an HDR video from an external monitor, toggle HDR mode on from an M2 pro macbook pro.

- [ ] MPV does not have this problem.

**How often does this happen?**
Every time i play an HDR video on my M2 pro macbook pro.

Just got this M2 pro a few days ago. I was enjoying HDR video in iina on my 2019 macbook before. Apple support was unable to fix the issue after 2 days as there were also issues with HDR distortion in all browsers besides safari. Fortunately, I saw that there was a makeshift fix in these forums that actually fixed both the browser and video players hdr issues. Was just wondering if there was a more definite fix today as the ones i read were from last year.

**Comment by low-batt** [comment]
The M2 & M3 problems are suggestive of a macOS defect as M1 does not exhibit these problems. So we have been hoping the problems would disappear in a macOS release.

That the workarounds work we can't explain either. Feels like there is some faulty graphics optimization in macOS that doesn't get applied when you change the environment. But this is all speculation.

Interesting to hear that Safari does not exhibit the problem, but other browsers do. I don't remember that being reported before.

Right at this moment IINA is working on upgrading `libmpv`. Possibly the [mpv](https://github.com/mpv-player/mpv) project has found a workaround or fix for this problem. We will have to see once we have a version of IINA working with updated dependencies.

**Comment by RoseAF1** [comment]
Makes sense. If it's of any interest to you or anyone with the same problem, I did have to enable hardware acceleration in browser for these makeshift fixes to work in any chromium browser (as i said, hdr already works in safari without problem). The fixes worked in IINA without touching anything within IINA so as you've said, it's definitely not an IINA issue.

**Comment by low-batt** [comment]
That hardware acceleration must be enabled for the workarounds to work in a chromium browser is another new puzzling piece of evidence. On a M chip hardware acceleration offloads decoding work from the CPUs to the Media Engine. Odd that doing so would enable the workarounds to work.
